File: cpp_programs/linkedlist.cpp
#include<iostream>
using namespace std;

struct Node{
    int num;
    Node* next;
};

struct Node* head = NULL;

void insert_node(int n){
    struct Node *new_node = new Node();
    new_node ->num = n;
    new_node ->next = head;
    head = new_node;
}

void display(){
    struct Node* temp = head;
    while (temp!=NULL){
        cout<<temp->num<<" ";
        temp = temp ->next;
    }

}

int main(){
    int n;
    cout<<"enter the number of test cases :";
    cin>>n;
    for(int i =0;i<n;i++){
        int x;
        cout<<"Enter element "<<i+1<<endl;
        cin>>x;
        insert_node(x);
    }
    cout<<"Given linked list is :";
    display();
    return 0;
}

File: LinkedList.py
# Python3 program to delete middle 
# of a linked list 

# Link list Node 
class Node: 
	
	def __init__(self): 
		
		self.data = 0
		self.next = None
	
# Count of nodes 
def countOfNodes(head): 

	count = 0
	
	while (head != None): 
		head = head.next
		count += 1
	
	return count 

# Deletes middle node and returns 
# head of the modified list 
def deleteMid(head): 

	# Base cases 
	if (head == None): 
		return None
	if (head.next == None): 
		del head 
		return None

	copyHead = head 

	# Find the count of nodes 
	count = countOfNodes(head) 

	# Find the middle node 
	mid = count // 2

	# Delete the middle node 
	while (mid > 1): 
		mid -= 1
		head = head.next

	# Delete the middle node 
	head.next = head.next.next

	return copyHead 

# A utility function to print 
# a given linked list 
def printList(ptr): 

	while (ptr != None): 
		print(ptr.data, end = '->') 
		ptr = ptr.next
	
	print('NULL') 
	
# Utility function to create a new node. 
def newNode(data): 

	temp = Node() 
	temp.data = data 
	temp.next = None
	return temp 

# Driver Code 
if __name__=='__main__': 
	
	# Start with the empty list 
	head = newNode(1) 
	head.next = newNode(2) 
	head.next.next = newNode(3) 
	head.next.next.next = newNode(4) 

	print("Given Linked List") 
	printList(head) 

	head = deleteMid(head) 

	print("Linked List after deletion of middle") 
	printList(head) 


